### ISSUE TITLE
GTK3: file dialogs are causing UI freeze (Antergos Linux)

### ISSUE THREAD

**Issue by bpasero** [electron/electron#12302]
* Electron version: 2.0.0-beta.3
* Operating system: Antergos Linux (4.15.9-1-ck)

### Expected behavior

File dialogs open normally and the UI does not freeze. 

### Actual behavior

File dialogs cause the window and menu to freeze. 

### How to reproduce

This was reported from @l0k1verloren via https://github.com/Microsoft/vscode/issues/45542#issuecomment-372916724

When the folder dialog opened nothing rendered and the menus didn't respond.

Initial startup:

![image](https://user-images.githubusercontent.com/20619963/37386247-474f06b0-2760-11e8-95fd-c02f28f91470.png)

After opening a (golang) repository folder:

![image](https://user-images.githubusercontent.com/20619963/37386283-60ba1ae0-2760-11e8-9344-a1917be0ceb0.png)

/cc @ckerr

**Comment by ckerr** [comment]
I'm not seeing this with `electron/2-0-x` (effectively 2.0.0-beta.6) + vscode `tyriar/electron-2.0.x` today.

I do see a period of unresponsiveness while the folder is being loaded (ie, after the folder was selected and the folder dialog is closed), and that _does_ seem to take slightly longer in the new build than in `vscode 1.21.1 + shell 1.7.9`, but it's not clear to me that's an Electron issue or that it's what @l0k1verloren was reporting, which sounds more like a full-stop.

I'm going to mark this as needsinfo for now but am very interested in getting electron 2 to play nicely with vscode. If this issue persists in electron-2.0.0-beta.6, please @ me in this ticket with followup information on how to test and what to look for. Thanks!

**Comment by l0k1verloren** [comment]
This issue seems to be resolved now, I have tested vscode with the electron 2 and it's working fine. There is a serious issue with the Nvidia driver that has emerged in the last few months (it was working until some time in late january or early february) and I fixed my issue by downgrading and I'm very reluctant to upgrade the nvidia driver on production machines until this issue is conclusively gone. I am not sure if this is relevant to this issue or not, but I strongly suspect it is. I just resolved an issue with frequent GPU related crashes on a Windows 10 machine yesterday also, so the current iterations of Nvidia drivers are definitely problematic, in a quite severe way.
